examples/printer/main.go

```diff
@@ -47,7 +47,7 @@ func main() {
 
 	// React to Ctrl+C signal, and stop gracefully after the first signal
 	// Second signal exits the process
-	signalC := make(chan os.Signal)
+	signalC := make(chan os.Signal, 2)
```

Comment:
I'd appreciate some comment (either in code or in commit message) what do those `2` and `3` mean.

Reply:
Explained offline. In a nutshell: `signal.Notify` does not block and may drop notification if the channel is full. The applications terminate after receiving 2/3 signals, so it does not make sense for those channels to be larger.